hydrosdk/cluster.py

```diff
@@ -1,40 +1,78 @@
+import json
 import logging
 from urllib import parse
-import requests
-import json
+
 import grpc
-import functools
+import requests
 
 
 class Cluster:
     """
     Cluster responsible for server interactions
     """
+
     @staticmethod
-    def connect(address):
+    def connect(http_address: str, grpc_address: str = None) -> 'Cluster':
         """
         The preferable factory method for Cluster creation. Use it.
 
-        :param address: connection address
-        :raises ConnectionError: if no connection with cluster
+        :param http_address: http connection address
+        :param grpc_address: optional grpc connection address
         :return: cluster object
 
         Checks the address, the connectivity, and creates an instance of Cluster.
         """
-        cl = Cluster(address)
-        logging.info("Connecting to {} cluster".format(cl.address))
-        info = cl.build_info()
+        cl = Cluster(http_address=http_address, grpc_address=grpc_address)
+
+        Cluster.check_connection(cluster=cl, address=http_address, address_type="http")
+        if grpc_address:
+            Cluster.check_connection(cluster=cl, address=grpc_address, address_type="grpc")
+
+        return cl
+
+    @staticmethod
+    def check_connection(cluster: 'Cluster', address: str, address_type: str) -> None:
+        """
+        Checks cluster connection
+        :param cluster: active cluster
+        :param address: address to check
+        :param address_type: grpc or http
+        :raises ConnectionError: if no connection with cluster
+        :return: None
+        """
+        logging.info("Connecting to {} cluster".format(address))
+
+        if address_type == "http":
+            info = cluster.build_info(manager=True, sonar=True, gateway=True)
+        elif address_type == "grpc":
+            info = cluster.build_info(manager=True, gateway=True)
```

Comment:
GPRC validation looks like it doesn't validate GPRC connectivity at all. Could you please elaborate on the validation here?

Reply:
@KineticCookie for a grpc connection to be made manager (cluster request) and gateway(manager request) should be running.

Reply:
but here you check connectivity of http protocol, not grpc

Reply:
Added connectivity check